0iq-tech/omnoma-mobile
# CI: Add Static Analysis

## Changes
- Configure development tooling (pre-commit hooks, lint-staged, commitlint)
- Add FSD architecture check workflow for PRs
- Update dependencies (steiger, steiger-plugin)

## Technical Details
- Added lint-staged + commitlint for commit validation
- Configured new GitHub workflow for FSD architecture checks
- Updated steiger packages to v0.5.2
- Removed standalone typecheck script (now in lint-staged)

## Screenshots
TBA

## Recordings
TBA

## Testing
1. Make changes and commit to verify:
   - Pre-commit hooks
   - Commit message format
   - FSD architecture rules
2. Create a PR to test the new workflow
